app/actions/chat.js

```diff
@@ -96,26 +97,20 @@ export function unsubscribe() {
 function handleMessage(dispatch, getState, payload) {
   const {id, from, message} = payload
 
-  const contactList: ContactList = getState().contactList
-  const contact = contactList.findContact(from)
-
-  if(!contact) {
-    console.log('Received message from unknow contact ' + from)
-    return
-  }
-
-  console.log('Received \'' + message + '\' from ' + contact.identity)
-  dispatch(priv.chatReceived(contact, id, message))
-  dispatch(sendChatAck(contact, id))
-
+  var contact = dispatch(forceFetchProfile(from,getState)).then(function (cont) {
```

Comment:
I get `Actions must be plain objects. Use custom middleware for async actions.` If i use that way. No idea why.